### PR TITLE
GitHub issue helpers

### PR DESCRIPTION
**Pull request by hadley**
First pass implements `issue_close_community()` and `issue_reprex_needed()`.

### CODE REVIEW

File: R/issue.R [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#' Helpers for GitHub issues
+#'
+#' @description
+#' The `issue_*` family of functions allows to perform common operations on
```
Comment (jennybc) [inline R/issue.R]:
```suggestion
#' The `issue_*` family of functions allows you to perform common operations on
```

File: R/issue.R [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+#' GitHub issues from within R. They're designed to help you efficiently deal
+#' with large numbers of issues, particularly motived but the challenges faced
```
Comment (jennybc) [inline R/issue.R]:
```suggestion
#' with large numbers of issues, particularly motivated by the challenges faced
```

File: R/issue.R [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+#' by the tidyverse team.
+#'
+#' * `issue_close_community()` closes an issue because it's not a bug report or
+#'   features, and points the author towards RStudio community for additional
```
Comment (jennybc) [inline R/issue.R]:
```suggestion
#'   feature request, and points the author towards RStudio community for additional
```

File: R/issue.R [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  )
+}
+
+issue_edit <- function(number, ...) {
+  issue_gh("PATCH /repos/:owner/:repo/issues/:issue_number", number, ...)
+}
+
+issue_info <- function(number) {
+  issue_gh("GET /repos/:owner/:repo/issues/:issue_number", number)
+}
```
Comment (jennybc) [inline R/issue.R]: I don't follow how `owner` and `repo` are making their way into these calls.

Comment (hadley) [reply R/issue.R]: `issue_gh()` assume that it's always the current repo + owner

Comment (jennybc) [reply R/issue.R]: D'oh. Somehow I read these as `gh::gh()` calls (vs. `issue_gh()`).

**Comment by jennybc** [comment]
Shall we merge?

**Comment by hadley** [comment]
I think let them marinate for a bit longer; I want to use in anger a bit first.

**Comment by jennybc** [comment]
What are your current feels on this @Hadley?

**Comment by hadley** [comment]
Yeah, I think it's fine to merge. Do you mind taking over the finish line?

**Comment by jennybc** [comment]
> Do you mind taking over the finish line?

OK will do.

**Pull request merged**